packages/loot-core/src/client/data-hooks/reports.ts

```diff
@@ -27,9 +27,9 @@ function toJS(rows: CustomReportData[]) {
       includeCurrentInterval: row.include_current === 1,
       showUncategorized: row.show_uncategorized === 1,
       graphType: row.graph_type,
-      conditions: row.conditions,
+      ...(row.conditions && { conditions: row.conditions }),
```

Comment:
Nit: Would this also work? This form is more readable IMO.
```suggestion
      conditions: row.conditions ? row.conditions : undefined,
```

Reply:
Unfortunately no, this rule makes it so you have to either fully omit the key, or allow the type to be undefined. So in this case, we either do the ugly spread thing, or we update `CustomReportEntity` to allow `conditions` to be `undefined`. Ideally I'd like to avoid making our model objects accept `undefined`, but defer to you